Kibana chart

jagregory opened the pull request:
Installing this chart creates a Deployment of Kibana, configured with a ConfigMap. I believe this satisfies the contributors guidelines, but it's my first chart so please let me know if there's anything missing/weird.

The chart defaults to Kibana 5.2, which is the latest version at the time of writing; however, this is unfortunately incompatible with the out-of-date incubator/elasticsearch. I figured it better to keep Kibana latest and allow downgrades, than pin Kibana based on an incubator chart.

jagregory committed:
[stable/kibana] Initial commit of a working Kibana chart

jagregory committed:
[stable/kibana] Use config map for generating kibana.yml

jagregory committed:
[stable/kibana] Readiness and liveness probes

seanknox commented:
@k8s-bot ok to test

seanknox commented:
Hi @jagregory, thanks for the chart and sorry for the delay in responding. It looks like this chart depends on elasticsearch—does that sound right?

```
$ kubectl logs -f --namespace default $(kubectl get pods --namespace default -l app=mottled-meerkat-kibana -o jsonpath='{ .items[0].metadata.name }')
{"type":"log","@timestamp":"2017-03-22T23:19:41Z","tags":["status","plugin:kibana@5.2.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-03-22T23:19:42Z","tags":["status","plugin:elasticsearch@5.2.0","info"],"pid":7,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-03-22T23:19:42Z","tags":["status","plugin:console@5.2.0","info"],"pid":7,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
{"type":"log","@timestamp":"2017-03-22T23:19:42Z","tags":["error","elasticsearch","admin"],"pid":7,"message":"Request error, retrying\nHEAD http://elasticsearch:9200/ => getaddrinfo ENOTFOUND elasticsearch elasticsearch:9200"}
{"type":"log","@timestamp":"2017-03-22T23:19:43Z","tags":["warning","elasticsearch","admin"],"pid":7,"message":"Unable to revive connection: http://elasticsearch:9200/"}
{"type":"log","@timestamp":"2017-03-22T23:19:43Z","tags":["warning","elasticsearch","admin"],"pid":7,"message":"No living connections"}
```

jagregory commented:
Yes, that's correct. Kibana is a web interface for querying/visualising elasticsearch.

Is there something I've missed?

seanknox commented:
Charts that have dependencies should declare them using a [requirements.yaml](https://github.com/kubernetes/helm/blob/master/docs/charts.md#managing-dependencies-with-requirementsyaml)—good explanation how the requirements system works at that link. The Drupal chart is a good example of using dependencies: https://github.com/kubernetes/charts/tree/master/stable/drupal

jagregory commented:
Thanks @seanknox. I'm not certain this is correct for this chart though, as I don't want elasticsearch to be installed as part of this chart. Users will already have an elasticsearch cluster setup, probably heavily customised, and Kibana is a view into it. To have this chart setup elasticsearch doesn't really make sense. It's akin to Grafana requiring Graphite, Grafana won't work without it but the Grafana chart shouldn't be setting Graphite up.

Does that make sense?

seanknox commented:
@jagregory That makes sense. One of the technical requirements for new charts is that all pods go to the running state with default values, OR provide further instructions. How about adding a note to that effect in `NOTES.txt`?

jagregory commented:
Thanks. This chart should go into running state as is, but it won't be very useful until you supply your own elasticsearch URL. I can certainly add something to NOTES.txt about updating the URL.

seanknox commented:
@jagregory the pods fail to start without ES (which can be OK as long as those expectations are in `NOTES.txt`:

```
$ k get po
NAME                                                           READY     STATUS     RESTARTS   AGE
dozing-bumblebee-kibana-1882671226-3f39f                       0/1       Running    31         1h
```

jagregory commented:
Hmm, ok I would've thought it should start up ok; usually it should come up but indicate it can't connect to ES.

I'll take a look and see why this might be. Failing that, I'll update the notes.

jagregory committed:
[stable/kibana] Change liveness and readiness probe urls

jagregory committed:
[stable/kibana] Notes for changing elasticsearch URL

jagregory commented:
@seanknox Looks like I got the readiness probe URLs wrong. I've fixed these up and now the chart starts without any changes. I've also added a note into NOTES.txt about changing the setting.

jagregory committed:
[stable/kibana] Add ingress

mstump commented:
Two issues with this chart in its current form:

The notes for this chart are incorrect. To set the ES url it should be `--set "config.elasticsearch.url"` not `--set "elasticsearch.url"`

Additionally, the config volume is being mounted to `/opt/kibana/config` and not `/etc/kibana` which is where the process is looking for config files.

mkrakowitzer commented:
To make this work I had to update the volume path as the current one was not read (https://www.elastic.co/guide/en/kibana/current/_configuring_kibana_on_docker.html)

It may also be worth noting the docker image used is from a deprecated repository and should be replaced by: docker.elastic.co/kibana/kibana (https://www.elastic.co/guide/en/kibana/current/_pulling_the_image.html)

```
--- a/stable/kibana/templates/deployment.yaml
+++ b/stable/kibana/templates/deployment.yaml
@@ -33,7 +33,7 @@ spec:
           initialDelaySeconds: 10
         volumeMounts:
         - name: config
-          mountPath: /opt/kibana/config
+          mountPath: /usr/share/kibana/config
         resources:
 {{ toYaml .Values.resources | indent 10 }}
       volumes:
```
its also not compatible with the elasticsearch chart in this repo, but this is more an issue with that chart as it should be updated to elasticsearch 5.x, its currently 2.4.0

viglesiasce commented:
@jagregory any updates? can you address the comments from @mkrakowitzer and @mstump?

jagregory commented:
@viglesiasce I'll look into it, thanks.

viglesiasce commented:
Closing as this has been stale for more than a week. Please re-open/re-submit when you have a chance to work on it.
https://github.com/kubernetes/charts#stale-pull-requests
